app/assets/javascripts/dropzone_image_upload.js

```diff
@@ -22,21 +25,29 @@
 
   var bindHandlers = function() {
     $('.campaign-images').on('ajax:success', "a[data-method=delete]", function(){
-      $(this).parents('.image-thumb').fadeOut();
+      $(this).parents('.dz-preview').fadeOut();
+      var imageId = $(this).parents('[data-image-id]').data('image-id');
+      $.publish('image:destroyed', imageId)
     });
   };
 
-  var checkAnyImages = function() {
-    if( $('.campaign-images img').length == 0) {
-      $('.campaign-images').hide();
-    }
-  };
-
   var initialize = function() {
     configureDropZone();
     bindHandlers();
   };
 
+  var addImageOption = function(e, file, id, html) {
+    var newOption = "<option value='"+id+"'>"+file.name+"</option>";
```

Comment:
Roger that. I'm going through this now and have noticed there's a lot of
different flavor JavaScript going on.

On Mon, Nov 2, 2015 at 3:30 PM, NealJMD <notifications@github.com> wrote:

> In app/assets/javascripts/dropzone_image_upload.js
> <https://github.com/SumOfUs/Champaign/pull/210#discussion_r43699202>:
>
> >    var initialize = function() {
> >      configureDropZone();
> >      bindHandlers();
> >    };
> >
> > +  var addImageOption = function(e, file, id, html) {
> > +    var newOption = "<option value='"+id+"'>"+file.name+"</option>";
>
> I have it using nice es6 string interpolation elsewhere and this was the
> low-bar downgrade. I'd rather just bring this into es6 I think...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/SumOfUs/Champaign/pull/210/files#r43699202>.
>